gpu-mg/MatrixCoverGPU.cu

```diff
@@ -2,20 +2,23 @@
 
 namespace gpu_mg {
 
+constexpr int size_bit = 1 << 31;
+
 __device__ void delete_rows_and_columns(int *dl_matrix, int *deleted_rows,
                                         int *deleted_cols,
                                         const int search_depth,
                                         const int selected_row_id,
                                         const int total_dl_matrix_row_num,
                                         const int total_dl_matrix_col_num) {
+  int selected_row_idx = selected_row_id * total_dl_matrix_col_num;
+
   for (int i = threadIdx.x; i < total_dl_matrix_col_num; i = i + blockDim.x) {
-    if (dl_matrix[selected_row_id * total_dl_matrix_col_num + i] == 1 &&
-        deleted_cols[i] == 0) {
+    if (deleted_cols[i] == 0 && dl_matrix[selected_row_idx + i] == 1) {
       deleted_cols[i] = search_depth;
-      for (int j = 0; j < total_dl_matrix_row_num; j++) {
-        if (dl_matrix[j * total_dl_matrix_col_num + i] == 1 &&
-            deleted_rows[j] == 0) {
-          atomicExch(deleted_rows + j, search_depth);
+      for (int j = 0; j < total_dl_matrix_row_num; ++j) {
+        if (deleted_rows[j] == 0 &&
+            dl_matrix[j * total_dl_matrix_col_num + i] == 1) {
+          deleted_rows[j] = search_depth;

```

Comment:
There might be cases when multiple columns want to delete one row. And multiple threads will write to the same device memory location. This does not affect results here, but we would still like to avoid such operations in the code?
Do you have any comments?